service/controller/resource/values/resource.go

```diff
@@ -0,0 +1,162 @@
+package values
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	vaultapi "github.com/hashicorp/vault/api"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/giantswarm/config-controller/pkg/decrypt"
+	"github.com/giantswarm/config-controller/pkg/generator"
+	"github.com/giantswarm/config-controller/pkg/github"
+)
+
+const (
+	Name = "values"
+)
+
+type Config struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+
+	VaultClient  *vaultapi.Client
+	GitHubToken  string
+	Installation string
+}
+
+type Resource struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+
+	decryptTraverser *decrypt.YAMLTraverser
+	gitHubToken      string
+	installation     string
+}
+
+func New(config Config) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.VaultClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.VaultClient must not be empty", config)
+	}
+	if config.GitHubToken == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GitHubToken must not be empty", config)
+	}
+	if config.Installation == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Installation must not be empty", config)
+	}
+
+	var decryptTraverser *decrypt.YAMLTraverser
+	{
+		c := decrypt.VaultDecrypterConfig{
+			VaultClient: config.VaultClient,
+		}
+
+		decrypter, err := decrypt.NewVaultDecrypter(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		decryptTraverser, err = decrypt.NewYAMLTraverser(
+			decrypt.YAMLTraverserConfig{
+				Decrypter: decrypter,
+			},
+		)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	r := &Resource{
+		k8sClient:        config.K8sClient,
+		logger:           config.Logger,
+		decryptTraverser: decryptTraverser,
+		gitHubToken:      config.GitHubToken,
+		installation:     config.Installation,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) GenerateConfig(ctx context.Context, owner, installation, namespace, app, configVersion string) (configmap *corev1.ConfigMap, secret *corev1.Secret, err error) {
+	var store generator.Filesystem
+	var ref string
+	{
+		gh, err := github.New(github.Config{
+			Token: r.gitHubToken,
+		})
+		if err != nil {
+			return nil, nil, microerror.Mask(err)
+		}
+
+		if configVersion == "" {
+			return nil, nil, microerror.Maskf(executionFailedError, "configVersion must be defined")
+		}
+
+		isTag, err := gh.ResolvesToTag(ctx, owner, app, configVersion)
```

Comment:
I think we can use `key.Owner` directly here. Or even make it `vaules` package scope const. It isn't probably gonna be reused between resources.